### PR TITLE
reward converter - OnChanOpenAck unit tests

### PR DESCRIPTION
**Pull request by sampocs**
## Brief Changelog
* Added unit tests for OnChanOpenAck
* Added helpers to keeper_test and test_helpers

### CODE REVIEW

**Review by ethan-stride — COMMENTED** [review]
LGTM
Still thinking about the weird edge cases of OnChanOpenAck itself, but I think these tests cover the intended flows... please get an additional focused review from V,A,or R on this very core but small piece of code for more confidence

File: x/stakeibc/keeper/ibc_test.go [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+		tradeAddress = types.ICAAccountType_CONVERTER_TRADE.String()
+	}
+
+	// Confirm the expected addresses with the host zone
+	tradeRoute, found := s.App.StakeibcKeeper.GetTradeRoute(s.Ctx, RewardDenom, HostDenom)
+	s.Require().True(found, "trade route should have been found")
+
+	s.Require().Equal(unwindAddress, tradeRoute.RewardAccount.Address, "unwind address")
+	s.Require().Equal(tradeAddress, tradeRoute.TradeAccount.Address, "trade address")
+}
```
[the remark this answers is not in the document]
Comment (sampocs) [reply x/stakeibc/keeper/ibc_test.go:233]: can you elaborate on what the change would be?

File: x/stakeibc/keeper/ibc_test.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+	err = s.App.StakeibcKeeper.OnChanOpenAck(s.Ctx, tradePortId, tradeChannelId)
+	s.Require().NoError(err, "no error expected when running callback with non-ICA port")
+
+	finalHostZone := s.MustGetHostZone(delegationChainId)
+	s.Require().Equal(hostZone, finalHostZone, "host zone should not have been modified")
+
+	finalTradeRoute, found := s.App.StakeibcKeeper.GetTradeRoute(s.Ctx, RewardDenom, HostDenom)
+	s.Require().True(found, "trade route should have been round")
+	s.Require().Equal(tradeRoute, finalTradeRoute, "trade route should not have been modified")
+}
```
[the remark this answers is not in the document]
Comment (sampocs) [reply x/stakeibc/keeper/ibc_test.go:81]: As discussed offline, I don't think we'd ever store down the withdrawal address in `StoreTradeRouteIcaAddress`, but lmk if you still think this is an issue

File: x/stakeibc/keeper/ibc_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ethan-stride) [reply x/stakeibc/keeper/ibc_test.go:233]: I was thinking before that we should include a setup test around withdrawal address because it goes through both the HostZone and it needs to show up on the TradeRoute -- but I don't think that it matters now since we assume the HostZone is always initialized first before the TradeRoute and we get the withdrawal address from the HostZone.  I no longer think this adds anything beyond the delegation address setup
